src/google/google-calendars/googlecalendarsyncadaptor.cpp

```diff
@@ -218,43 +403,58 @@ void extractRecurrence(const QJsonArray &recurrence, KCalCore::Event::Ptr event,
         return;
     }
 
+    kcalRecurrence->clear(); // avoid adding duplicate recurrence information
```

Comment:
Is that above unsetRecurs() check valid? Looks to me like this might leave some RDATEs behind when other end clears all recurrence state.